03_letters/sirigithub/main_test.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLetterFrequency(t *testing.T) {
+	tests := []struct {
+		description string
+		input       string
+		expected    []string
+	}{
+		{description: "Simple string", input: "thisisasimplestring",
+			expected: []string{"a:1", "e:1", "g:1", "h:1", "i:4", "l:1", "m:1", "n:1", "p:1", "r:1", "s:4", "t:2"}},
+
+		{description: "String with spaces", input: "   223  ", expected: []string{" :5", "2:2", "3:1"}},
+
+		{description: "Empty String", input: "", expected: []string{}},
+
+		{description: "German Umlauts", input: "äöß€’üüüöäßß", expected: []string{"ß:3", "ä:2", "ö:2", "ü:3", "’:1", "€:1"}},
+
+		{description: "String with Back slashes", input: "a\\c\\b", expected: []string{"\\:2", "a:1", "b:1", "c:1"}},
+
+		{description: "String with Emojis", input: "😄🐷🙈🐷🏃😄", expected: []string{"🏃:1", "🐷:2", "😄:2", "🙈:1"}},
+	}
+
+	for _, test := range tests {
+		actual := sortLetters(letters(test.input))
+		fmt.Println(actual)
```

Comment:
It's best not to print things out in tests. On a large program, if tests were to print out stuff you'd end up with a mess of output, particularly as tests can run concurrently - all the output would be interleaved.

Also, in the case of this repo, once this is merged to master, this test is going to run every time and this output will be in the common travis output. So best to remove this before merging.

It can be handy having these prints there when writing/growing your tests, but please be sure to remove it when you push your commits.